force-app/main/default/classes/DMLOperations.cls

```diff
@@ -146,9 +176,34 @@ public with sharing class DMLOperations {
      * @param oppNames The list of Opportunity names.
      */
     public static void upsertOpportunities(String accountName, List<String> oppNames) {
+        List<Opportunity> opportunities = new List<Opportunity>();
+        List<String> existingOpportunities = new List<String>();
+        Id accountId;
+        for(Opportunity existingOpportunity : [SELECT Name FROM Opportunity WHERE Name IN :oppNames AND Account.Name = :accountName]){
```

Comment:
You seem to be querying the opportunity with account name before you create the accountName if it doesn't exist